internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -135,5 +149,62 @@ func (r *Receiver) createMonitor(monitorType string) (interface{}, error) {
 		output.AddExtraDimension(k, v)
 	}
 
-	return monitor, nil
+	// Configure SmartAgentConfigProvider to gather any global config overrides and
+	// set required envs.
+	configureEnvironmentOnce.Do(func() {
+		r.setUpSmartAgentConfigProvider(extensions)
+		setUpEnvironment()
+	})
+
+	// Note, that this receiver has to configure main collectd even for collectd/custom monitor,
+	// despite the fact that, that monitor stands up its own instance of collectd to prevent this
+	// panic "Main collectd instance should not be accessed before being configured".
```

Comment:
nit, not sure this comment is necessary or too informative since we don't really do any monitor type-based logic.